### DISCUSSION TITLE
TODO

### DISCUSSION THREAD

**Discussion by cieslarmichal** [cieslarmichal/faker-cxx#4] [General]
Let's write next features to implement here.

**Comment by cieslarmichal** [comment]
Next steps:
 - Phone: Generating phone number
 - Location: Generate valid Addresses, Zip Codes, Street Names, States, and Countries
 - Lorem: Generating lorem sentences
 - Internet: Generating avatar link, color, IPV4, IPV6, emoji, HttpMethodName, HttpStatusCode, port, protocol, url
 - Add github documentation about available functions

**Reply by dario-loi** [reply]
The functionality in helper should be provided by the standard library header algorithm, hence we could just build a thin wrapper around it like number, but I doubt there are any benefits to it compared to the standard approach.

**Reply by cieslarmichal** [reply]
Yeah you probably right, lets drop the shuffling. (deleted from TODO list)

**Reply by cieslarmichal** [reply]
I finished date class yesterday, but need to write docs to it. I think it would be good idea to write example generated data in our function documentationx like they do in fakerjs:

```
  /**
   * Generates an email address using the given person's name as base.
   *
   * @param options The options to use. Defaults to `{}`.
   * @param options.firstName The optional first name to use. If not specified, a random one will be chosen.
   * @param options.lastName The optional last name to use. If not specified, a random one will be chosen.
   * @param options.provider The mail provider domain to use. If not specified, a random free mail provider will be chosen.
   * @param options.allowSpecialCharacters Whether special characters such as ``.!#$%&'*+-/=?^_`{|}~`` should be included
   * in the email address. Defaults to `false`.
   *
   * @example
   * faker.internet.email() // 'Kassandra4@hotmail.com'
   * faker.internet.email({ firstName: 'Jeanne', lastName: 'Doe' }) // 'Jeanne63@yahoo.com'
   * faker.internet.email({ firstName: 'Jeanne', lastName: 'Doe', provider: 'example.fakerjs.dev' }) // 'Jeanne_Doe88@example.fakerjs.dev'
   * faker.internet.email({ firstName: 'Jeanne', lastName: 'Doe', provider: 'example.fakerjs.dev', allowSpecialCharacters: true }) // 'Jeanne%Doe88@example.fakerjs.dev'
   *
   * @since 2.0.1
   */
```

**Reply by cieslarmichal** [reply]
I created documentation in Date class fyi

**Reply by dario-loi** [reply]
Great idea on the examples and GJ on the docs, it could be worth it to set up CI/CD to autodeploy the docs to github pages through doxygen (perhaps with a CSS template to make it less ugly).

I am still on the float module and I'll probably be done tomorrow, since I cant work on it today.

**Reply by cieslarmichal** [reply]
Ok I will look into that autodeploy docs to github pages today :)

**Reply by cieslarmichal** [reply]
Added to issues about lib dependencies, because  couple of people on Reddit told me they would use this library if it would have no dependencies.

**Reply by dario-loi** [reply]
Just open up a PR with the floating point numbers, the changes are a bit hardcore (might have gone a bit overboard with the concepts), so feel free to trim it down if you want.

**Reply by cieslarmichal** [reply]
I see, yeah a bit hardcore, will take a look in a moment, would you like to take a look at Doxygen generation? I generated something in html directory but it's probably garbage? I will delete it and maybe we will try again.

**Reply by cieslarmichal** [reply]
I will switch from doxygen to date refactor to chrono

**Reply by dario-loi** [reply]
I checked that out, it seems to be working for number.h but other files such as Date do not show up, moreover I think the examples are not supposed to be written that way (?), as they seem to be completely wrong.

I've never written any examples on doxygen so idk about the right syntax, might check it out tomorrow as a next step.

**Reply by cieslarmichal** [reply]
ok thanks, I am checking your changes right now

**Reply by cieslarmichal** [reply]
good job! :fire:

**Reply by dario-loi** [reply]
Thank you mate, happy to help! what do you think will be the best course of action now? should I work on a simple module while you handle the docs?

**Reply by cieslarmichal** [reply]
So I think most important things to do are:
- dropping boost dependency (I am handling this)
- doxygen and automatic deploy to github pages
- setting GTest dependency as optional in CMake to give user and option whether they want to build our library with tests or not 
- phone module (generating phone number)
- location (generating country, city, zip code, state, street address etc)

You can choose what you would like to work on now :rocket:

**Reply by cieslarmichal** [reply]
I managed to fix @example in code, I replaced them with @code ... and @endcode after, works very well in locally generated doxygen :smile:

**Reply by cieslarmichal** [reply]
I am working on removing boost dependencies from Date module right now fyi (will be done today/tommorow).

**Reply by cieslarmichal** [reply]
done: https://github.com/cieslarmichal/faker-cxx/pull/21

**Comment by abhikumar63** [comment]
Hey,
I find this project interesting and want to contribute. I am new to open source bu I have experience in C++ language. I have got the gist of the file structure and programs. If you could guide me to how to contribute with my limited experience it would help me very much.

**Reply by cieslarmichal** [reply]
Hello! nice to meet you, here is a guide about contribution in general: [first-contributions](https://github.com/firstcontributions/first-contributions), you need to fork a repository and work on your local copy of repo, push changes to local repo and then create a pull request to main project repo.

I think a good place to start would be an Internet class, you could implement one of methods: ipv4, url, port, protocol or httpMethodName.

There is a fakerjs internet module out there if you want some guidance: [internet module](https://github.com/faker-js/faker/blob/next/src/modules/internet/index.ts).

So you could just pick ipv4 (or other function) implement this (four random numbers with dots), write unit tests covering this scenario, and create a pull request.

If you have any question, you can ask here, I am happy to help.

PS, don't forget to give a star to our library :fire:

**Comment by cieslarmichal** [comment]
I have just created [Discord Channel](https://discord.com/invite/jPy9R64Q) for contributors.
